projects/extension/src/background/index.ts

```diff
@@ -24,9 +24,46 @@ export interface Background extends Window {
       listener: (state: ExposedChainConnection[]) => void,
     ) => () => void
     disconnectTab: (tabId: number) => void
+    getLogger: () => LogKeeper
   }
 }
 
+interface LogKeeper {
+  debug: string[]
+  warn: string[]
+  error: string[]
+}
+
+const logKeeper: LogKeeper = {
+  debug: [],
+  warn: [],
+  error: [],
+}
+
+const getTime = () => {
+  const date = new Date()
+  return `${date.getHours() < 10 ? "0" : ""}${date.getHours()}:${
+    date.getMinutes() < 10 ? "0" : ""
+  }${date.getMinutes()}:${
+    date.getSeconds() < 10 ? "0" : ""
+  }${date.getSeconds()} ${date.getMilliseconds()}`
+}
+
+const logger = (level: number, target: string, message: string) => {
+  const logLevels = ["Error", "Warn", "Info", "Debug"]
+  if (level !== 4) {
+    // log all non-debug logs to background console
+    console.log(`${logLevels[level - 1]}: ${message}`)
```

Comment:
I know it's a bit annoying, but I'd prefer something like

```suggestion
    switch (level) {
        case 0: case 1: console.error(...); break;
        case 2: console.warn(...); break;
        case 3: console.info(...); break;
        case 4: console.log(...); break;
    }
```

So that we keep the proper level

Reply:
Is there a `case 0:`

Reply:
No normally there can't be `0`, but in case there is it's better to not just discard it

Reply:
fwiw the suggestion is not entirely correct. I mean that case 4 will not be logged (as we do not want `debug` to appear in the browser console. I will alter accordingly